packages/avatars/src/styled/StyledAvatar.js

```diff
@@ -0,0 +1,274 @@
+      `}
+  `;
+};
+
+const colorStyles = props => {
+  let statusColor = 'transparent';
+  const backgroundColor = props.backgroundColor || 'transparent';
+  const foregroundColor = props.foregroundColor || props.theme.palette.white;
+  const surfaceColor = props.status
+    ? props.surfaceColor || props.theme.colors.background
+    : 'transparent';
+
+  if (props.status === STATUS.AVAILABLE) {
+    statusColor = getColor('mint', 400, props.theme);
+  } else if (props.status === STATUS.ACTIVE) {
+    statusColor = getColor('crimson', 400, props.theme);
+  } else if (props.status === STATUS.AWAY) {
+    statusColor = getColor('yellow', 400, props.theme);
+  }
+
+  return css`
+    box-shadow: ${props.theme.shadows.sm(statusColor)};
+    background-color: ${backgroundColor};
+    color: ${surfaceColor};
+
+    & > svg,
+    & ${StyledText} {
+      color: ${foregroundColor};
+    }
+
+    &::after {
+      background-color: ${statusColor};
+      /* set text color without altering border */
+      -webkit-text-fill-color: ${props.theme.palette.white};
+    }
+
+    /* stylelint-disable selector-type-no-unknown, selector-no-vendor-prefix */
+    _:-ms-input-placeholder,
+    &::after {
+      /* fallback for IE11 (which will set border) */
+      color: ${props.theme.palette.white};
+    }
+    /* stylelint-enable selector-type-no-unknown, selector-no-vendor-prefix */
+  `;
+};
+
+const sizeStyles = props => {
+  let borderRadius;
+  let size;
+  let fontSize;
+  let svgSize;
+
+  if (props.size === SIZE.EXTRASMALL) {
+    borderRadius = props.isSystem ? math(`${props.theme.borderRadii.md} - 1`) : '50%';
+    size = math(`${props.theme.space.base} * 6px`);
+    fontSize = props.theme.fontSizes.sm;
+    svgSize = math(`${props.theme.space.base} * 3px`);
+  } else if (props.size === SIZE.SMALL) {
+    borderRadius = props.isSystem ? math(`${props.theme.borderRadii.md} - 1`) : '50%';
+    size = math(`${props.theme.space.base} * 8px`);
+    fontSize = props.theme.fontSizes.md;
+    svgSize = math(`${props.theme.space.base} * 3px`);
+  } else if (props.size === SIZE.LARGE) {
+    borderRadius = props.isSystem ? math(`${props.theme.borderRadii.md} + 1`) : '50%';
+    size = math(`${props.theme.space.base} * 12px`);
+    fontSize = props.theme.fontSizes.xl;
+    svgSize = math(`${props.theme.space.base} * 6px`);
+  } else {
+    borderRadius = props.isSystem ? props.theme.borderRadii.md : '50%';
+    size = math(`${props.theme.space.base} * 10px`);
+    fontSize = props.theme.fontSizes.lg;
+    svgSize = math(`${props.theme.space.base} * 4px`);
+  }
+
+  return css`
+    border-radius: ${borderRadius};
+    width: ${size};
+    height: ${size};
+
+    & > svg {
+      font-size: ${svgSize};
+    }
+
+    & ${StyledText} {
+      line-height: ${size};
+      font-size: ${fontSize};
+    }
+  `;
+};
+
+/**
+ * Accepts all `<figure>` props
+ */
+export const StyledAvatar = styled.figure.attrs(() => ({
```

Comment:
If we aren't interacting with props I think you can just return an object within `attrs`